### PR TITLE
Support for SPATIAL keys, GEOMETRY SRID

### PR DESCRIPTION
**Pull request by shlomi-noach**
This PR adds support for `SPATIAL` types and keys:

- `tengo` can now read `SRID` for spatial data types (supported via MySQL >= 8.0.3, no support at this time for MariaDB flavors)
- `tengo` outputs `SRID` in column definition, e.g. it generates the following, same as MySQL does
```sql
COLUMN `origin_g` GEOMETRY NOT NULL /*!80003 SRID 4326 */
```
- `tengo` correctly diffs tables with `SPATIAL` columns and keys. It is able to generate a `ADD SPATIAL KEY...` diff.

Some notes:

- MySQL 8.0.3 adds GRID support: https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-3.html#mysqld-8-0-3-spatial-support.
- `GRID` is available as column `SRS_ID` in `INFORMATION_SCHEMA.COLUMNS`
- In MySQL a `SPATIAL` key is listed as having `sub_part` value of `32` in `INFORMATION_SCHEMA.STATISTICS`. However, the `sub_part` is not actually used, and so we ignore it even when defined
- This PR made no research into MariaDB `SPATIAL` types and keys

### Contribution

This repo is a fork of https://github.com/skeema/tengo. Per current [CONTRIBUTING.md](https://github.com/skeema/tengo/blob/7d20c8c9a685dd72cc0e81f3fc9f9b18b26f4a5c/CONTRIBUTING.md) guidelines:

> Pull requests
> Thank you for your interest in contributing to this package. As of March 2021, most pull requests are not being accepted for this repo until further notice. At this time the maintainer of this package is focusing development energy primarily on commercial products, as this is necessary for open source development to resume in earnest in the future.

> Exceptions may be made in a few specific areas:
...
> Spatial index introspection and diff logic

I seems like `SPATIAL` support can be a welcome contribution. Here are patch files, to be applied with:

```
git am 0001-Support-for-SPATIAL-keys.patch.txt
git am 0002-beautify-CREATE-TABLE-statement.patch.txt
```
I'm happy if upstream wishes to incorporate this contribution. Please advise if you prefer a pull request.

[0001-Support-for-SPATIAL-keys.patch.txt](https://github.com/planetscale/tengo/files/6855318/0001-Support-for-SPATIAL-keys.patch.txt)
[0002-beautify-CREATE-TABLE-statement.patch.txt](https://github.com/planetscale/tengo/files/6856141/0002-beautify-CREATE-TABLE-statement.patch.txt)

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Fixes https://github.com/skeema/skeema/issues/97 for MySQL flavor

File: tengo_test.go [hunk 1]
```diff
@@ -327,6 +327,65 @@ func anotherTableForFlavor(flavor Flavor) Table {
...
+		Type: "SPATIAL",
+	}
+	stmt := `CREATE TABLE ` + "`" + `actor_spatial` + "`" + ` (
+  ` + "`" + `actor_id` + "`" + ` smallint(5) unsigned NOT NULL,
+  ` + "`" + `latlon_p` + "`" + ` point NOT NULL,
+  ` + "`" + `latlon_g` + "`" + ` geometry NOT NULL srid 4326,
+  PRIMARY KEY (` + "`" + `actor_id` + "`" + `),
+  SPATIAL KEY (` + "`" + `latlon_p` + "`" + `),
+  SPATIAL KEY (` + "`" + `latlon_g` + "`" + `),
+) ENGINE=InnoDB DEFAULT CHARSET=latin1`
```
Comment (vmg) [inline tengo_test.go:371]: Silly nit but if you simply quote everything here with `"` you can simplify the declaration a lot and remove many of the `+` operators. There's no reason why `CREATE TABLE ` needs to be quoted with backticks.

**Review by vmg — APPROVED** [review]

File: tengo_test.go (hunk 1 above)
Comment (shlomi-noach) [reply tengo_test.go:371]: I have merely complied with the coding convention used in similar tests. I do not know why this is the convention used, but I think it's best to stick to it if this is to be contributed upstream.

Comment (shlomi-noach) [reply tengo_test.go]: FWIW the reason this was done this way is that `"` does not support line breaks. However, further down the test file there was another method used, which I adopted.

**Pull request merged**